test/e2e/e2e.go

```diff
@@ -742,6 +727,133 @@ var _ = Describe("validation", func() {
 			}, timeout, interval).Should(BeTrue())
 		})
 	})
+	Context("Validate AddressPool Webhook", func() {
+		BeforeEach(func() {
+			By("Checking if validation webhook is enabled")
+			deploy, err := testclient.Client.Deployments(OperatorNameSpace).Get(context.Background(), consts.MetalLBOperatorDeploymentName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			isValidationWebhookEnabled := false
+			for _, container := range deploy.Spec.Template.Spec.Containers {
+				if container.Name == "manager" {
+					for _, env := range container.Env {
+						if env.Name == "ENABLE_OPERATOR_WEBHOOK" {
+							if env.Value == "true" {
+								isValidationWebhookEnabled = true
+							}
+						}
+					}
+				}
+			}
+
+			if !isValidationWebhookEnabled {
+				Skip("AddressPool webhook is disabled")
+			}
+
+			By("Checking if validation webhook is running")
+			// Can't just check the ValidatingWebhookConfiguration name as it's changing between different deployment methods.
+			// Need to check the webhook name in the webhooks definition of the ValidatingWebhookConfiguration.
+			validateCfgList := &admv1.ValidatingWebhookConfigurationList{}
+			err = testclient.Client.List(context.TODO(), validateCfgList, &goclient.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			isValidationWebhookRunning := false
+			for _, validateCfg := range validateCfgList.Items {
+				for _, webhook := range validateCfg.Webhooks {
+					if webhook.Name == consts.AddressPoolValidationWebhookName {
+						isValidationWebhookRunning = true
+					}
+				}
+			}
+			Expect(isValidationWebhookRunning).To(BeTrue(), "AddressPool webhook is not running")
+		})
+		It("Should recognize duplicate definition of AddressPools", func() {
```

Comment:
Should we split cases to different Its - duplicate addresspool names, overlapping addresses, ... ?
Are there more cases that are enforced by the webhook and justify being tested? CIDR that isn't parsable / ... ?

Reply:
I'd have a couple of broad e2e tests that verify that the webhook is working propertly, and extend the coverage of the logic as unit tests.